tests/Persistence/SqlAllMappingsLookupTest.php

```diff
@@ -0,0 +1,230 @@
+use Wikibase\Repo\WikibaseRepo;
+
+/**
+ * @covers \ProfessionalWiki\WikibaseRDF\Persistence\SqlAllMappingsLookup
+ * @group Database
+ */
+class SqlAllMappingsLookupTest extends \MediaWikiIntegrationTestCase {
+
+	protected function setUp(): void {
+		parent::setUp();
+
+		$this->setAllowedPredicates( [ 'foo:foo', 'foo:bar', 'foo:baz' ] );
+
+		$this->createItemWithMappings( new ItemId( 'Q99001' ) );
+		$this->createItemWithMappings(
+			new ItemId( 'Q99002' ),
+			new MappingList( [
+				new Mapping( 'foo:foo', 'https://example.com/#foo1' )
+			] )
+		);
+		$this->createItemWithMappings(
+			new ItemId( 'Q99003' ),
+			new MappingList( [
+				new Mapping( 'foo:foo', 'https://example.com/#foo1' ),
+				new Mapping( 'foo:bar', 'https://example.com/#bar1' ),
+				new Mapping( 'foo:baz', 'https://example.com/#baz1' )
+			] )
+		);
+		$this->createPropertyWithMappings( new PropertyId( 'P99001' ) );
+		$this->createPropertyWithMappings(
+			new PropertyId( 'P99002' ),
+			new MappingList( [
+				new Mapping( 'foo:foo', 'https://example.com/#foo1' )
+			] )
+		);
+		$this->createPropertyWithMappings(
+			new PropertyId( 'P99003' ),
+			new MappingList( [
+				new Mapping( 'foo:foo', 'https://example.com/#foo1' ),
+				new Mapping( 'foo:bar', 'https://example.com/#bar1' ),
+				new Mapping( 'foo:baz', 'https://example.com/#baz1' )
+			] )
+		);
+	}
+
+	/**
+	 * @param string[] $predicates
+	 */
+	private function setAllowedPredicates( array $predicates ): void {
+		$this->setMwGlobals( 'wgWikibaseRdfPredicates', $predicates );
+	}
+
+	private function createItemWithMappings( ItemId $itemId, ?MappingList $mappingList = null ): void {
+		WikibaseRepo::getEntityStore()->saveEntity(
+			new Item( $itemId ),
+			'',
+			self::getTestSysop()->getUser()
+		);
+
+		if ( $mappingList != null ) {
+			$this->setMappings( $itemId, $mappingList );
+		}
+	}
+
+	private function createPropertyWithMappings( PropertyId $propertyId, ?MappingList $mappingList = null ): void {
+		WikibaseRepo::getEntityStore()->saveEntity(
+			new Property( $propertyId, null, 'testType' ),
+			'',
+			self::getTestSysop()->getUser()
+		);
+
+		if ( $mappingList != null ) {
+			$this->setMappings( $propertyId, $mappingList );
+		}
+	}
+
+	private function setMappings( EntityId $entityId, MappingList $mappingList ): void {
+		WikibaseRdfExtension::getInstance()->newMappingRepository( self::getTestSysop()->getUser() )->setMappings(
+			$entityId,
+			$mappingList
+		);
+	}
+
+	public function newSqlAllMappingsLookup(): SqlAllMappingsLookup {
+		return new SqlAllMappingsLookup(
+			$this->db,
+			WikibaseRdfExtension::SLOT_NAME,
+			WikibaseRepo::getEntityIdParser(),
+			WikibaseRdfExtension::getInstance()->newMappingListSerializer()
+		);
+	}
+
+	public function testUnmodifiedMappings(): void {
+		$allMappings = $this->newSqlAllMappingsLookup()->getAllMappings();
+
+		// TODO: database is polluted by other tests
+		// $this->assertCount( 4, $allMappings );
```

Comment:
I couldn't find a quick way to reset the DB. At the moment our other tests are also creating mappings, so it's a bit difficult to make sure "all mappings" don't contain unexpected items.